File: BackEnd/backend/src/main/java/com/fitapp/backend/application/dto/exercise/response/ExerciseResponse.java
package com.fitapp.backend.application.dto.exercise.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fitapp.backend.infrastructure.persistence.entity.enums.ExerciseType;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@Schema(description = "Respuesta de ejercicio")
public class ExerciseResponse {

    @JsonProperty("id")
    private Long id;

    @JsonProperty("name")
    private String name;

    @JsonProperty("description")
    private String description;

    @JsonProperty("exerciseType")
    private ExerciseType exerciseType;

    @JsonProperty("sports")
    private Map<Long, String> sports = new HashMap<>();

    @JsonProperty("createdById")
    private Long createdById;

    @JsonProperty("categoryIds")
    private Set<Long> categoryIds = new HashSet<>();

    @JsonProperty("categoryNames")
    private Set<String> categoryNames = new HashSet<>();

    @JsonProperty("supportedParameterIds")
    private Set<Long> supportedParameterIds = new HashSet<>();

    @JsonProperty("supportedParameterNames")
    private Set<String> supportedParameterNames = new HashSet<>();

    @JsonProperty("isActive")
    private Boolean isActive;

    @JsonProperty("isPublic")
    private Boolean isPublic;

    @JsonProperty("usageCount")
    private Integer usageCount;

    @JsonProperty("rating")
    private Double rating;

    @JsonProperty("ratingCount")
    private Integer ratingCount;

    @JsonProperty("createdAt")
    private LocalDateTime createdAt;

    @JsonProperty("updatedAt")
    private LocalDateTime updatedAt;

    @JsonProperty("lastUsedAt")
    private LocalDateTime lastUsedAt;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/response/exercise/response/ExerciseResponse.kt
package com.fitapp.appfit.response.exercise.response

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class ExerciseResponse(
    @SerializedName("id")
    val id: Long,

    @SerializedName("name")
    val name: String,

    @SerializedName("description")
    val description: String?,

    @SerializedName("exerciseType")
    val exerciseType: ExerciseType?,

    @SerializedName("sportId")
    val sportId: Long?,

    @SerializedName("sportName")
    val sportName: String?,

    @SerializedName("createdById")
    val createdById: Long?,

    @SerializedName("createdByEmail")
    val createdByEmail: String?,

    @SerializedName("categoryIds")
    val categoryIds: Set<Long> = emptySet(),

    @SerializedName("categoryNames")
    val categoryNames: Set<String> = emptySet(),

    @SerializedName("supportedParameterIds")
    val supportedParameterIds: Set<Long> = emptySet(),

    @SerializedName("supportedParameterNames")
    val supportedParameterNames: Set<String> = emptySet(),

    @SerializedName("isActive")
    val isActive: Boolean?,

    @SerializedName("isPublic")
    val isPublic: Boolean?,

    @SerializedName("usageCount")
    val usageCount: Int?,

    @SerializedName("rating")
    val rating: Double?,

    @SerializedName("ratingCount")
    val ratingCount: Int?,

    @SerializedName("createdAt")
    val createdAt: String?,

    @SerializedName("updatedAt")
    val updatedAt: String?,

    @SerializedName("lastUsedAt")
    val lastUsedAt: String?
) : Serializable